test/models/job_queue_test.rb

```diff
@@ -94,6 +94,23 @@ def locked
       end
     end
 
+    it 'queues a job if max concurrent jobs is hit' do
+      with_job_execution do
+        locked do
+          JobQueue.concurrency = 1
```

Comment:
need to unset that in the `ensure`

Reply:
something like:

```
begin
  test
ensure
  ... unset ...
end
```

but as discussed above if you can do this via an instance method then we don't need to change global state at all